tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -266,9 +266,11 @@ def run(self):
                     ).load_all() or []
                     volumes = []
                     for volume in all_volumes:
-                        if not str(volume.deleted).lower() == "true" or \
+                        if not str(volume.deleted).lower() == "true" and \
                             volume.current_job.get('status', '') \
-                            in ['', 'finished', 'failed']:
+                            in ['', 'finished', 'failed'] and \
+                            volume.vol_id not in [None, ''] and \
+                            volume.name not in [None, '']:
                             volumes.append(volume)
```

Comment:
@nthomas-redhat if volume details are not populated and volume option details are populated then volume object will be empty, the check volume.vol_id not in [None, ''] and volume.name not in [None, ''] will takecare of omitting the volume empty volume obj.

Reply:
ack